android/src/main/java/com/reactnativestripesdk/StripeSdkModule.kt

```diff
@@ -172,7 +172,9 @@ class StripeSdkModule(private val reactContext: ReactApplicationContext) : React
   private val googlePayReceiver: BroadcastReceiver = object : BroadcastReceiver() {
     override fun onReceive(context: Context?, intent: Intent) {
       if (intent.action == ON_GOOGLE_PAY_FRAGMENT_CREATED) {
-        googlePayFragment = (currentActivity as AppCompatActivity).supportFragmentManager.findFragmentByTag("google_pay_launch_fragment") as GooglePayFragment
+        (currentActivity as? AppCompatActivity)?.let {
```

Comment:
I was really surprised that you have to check the current activity for null everywhere, even from @ReactMethod, but I didn't really see another way.

In order to prevent this type of error in the future I would recommend:
```
class StripeSdkModule ...{
...
  // Set currentActivity so it uses this instead of the base class `getCurrentActivity()`
  // this will give you type safety checking and convert it to an appCompat activity
  val currentActivity: AppCompatActivity?
    get() = (super.getCurrentActivity() as? AppCompatActivity)
...
    private val googlePayReceiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent) {
          if (intent.action == ON_GOOGLE_PAY_FRAGMENT_CREATED) {
            currentActivity?.let {
...
}
```


It would also be nice to see some enforcement in the code that if you can't get the appCompat activity that a promise is fired to return a result.

Reply:
thanks for the feedback! created a helper method for getting the activity or resolving the promise. It's not the perfect solution since someone can still call `currentActivity`, but it does avoid a lot of repeated code and hopefully anyone working here would see the current pattern for getting the activity 🤞

Reply:
@michelleb-stripe do you approve the most recent changes? I'll need to rebase but since https://github.com/stripe/stripe-react-native/pull/879 contains substantial changes to the same file, i'd rather wait till after that lands to rebase (to avoid duplicating the work)